### PR TITLE
Chrome: Unselect blocks when opening the document settings

### LINKED ISSUE #6887: Selecting the Document tab should deselect blocks.
**Describe the bug**

When you select a block and the Settings sidebar is open, the Block tab is automatically selected. 

When you deselect a block, the Document tab is then opened.

If you select a block, then click the Document tab, logically this should also deselect the block.

**To Reproduce**

- Click a block, observe the Block tab is opened
- Click the Documents tab, observe the block is not deselected

**Expected behavior**

Way back, in version 0.6 I think, we used something a similar feature to autoselect the block inspector when a block had focus. We used a "breadcrumb" style instead of tabs for the sidebar, the idea being that the sidebar was always contextual, and when you had no block selected you were at the document scope, when a block selected you dug into the block scope.

This behavior is now reinstated, although with tabs. But because deselecting a block automatically selects the Document tab, it feels like selecting the Document tab when a block is selected should logically also deselect the block. 

**Screenshots**
deselect

### PR DESCRIPTION
**Pull request by youknowriad**
closes #6887

**Testing instructions**

 - Select a block
 - This shows the block inspector
 - Click "Document"
 - This unselects the block

### CODE REVIEW

**Review by jasmussen — APPROVED** [review]
Nice! Works as intended. 

I can't review the code, but the experience feels right 👍 👍

**Review by gziolo — APPROVED** [review]
👍

File: edit-post/components/sidebar/settings-header/index.js [hunk 1]
```diff
@@ -42,7 +42,17 @@ export default compose(
...
 		count: select( 'core/editor' ).getSelectedBlockCount(),
 	} ) ),
-	withDispatch( ( dispatch ) => ( {
-		openSidebar: dispatch( 'core/edit-post' ).openGeneralSidebar,
-	} ) ),
+	withDispatch( ( dispatch ) => {
+		const { openGeneralSidebar } = dispatch( 'core/edit-post' );
+		const { clearSelectedBlock } = dispatch( 'core/editor' );
+		return {
+			openDocumentSettings: () => {
```
Comment (gziolo) [inline edit-post/components/sidebar/settings-header/index.js]: You can use object function shorthand:
```js
return {
	openDocumentSettings() { ... },
	openBlockSettings() { ... },
};
```

**Comment by afercia** [comment]
@youknowriad quick question: would it be easy to re-select the previously selected block when clicking "Block" after "Document" has been clicked?

**Comment by youknowriad** [comment]
@afercia We'd have to store this information somewhere, we don't have it at the moment.

**Pull request merged**